141617729 persisting food jc

jdconrad89 opened the pull request:
@case-eee @icorson3 @DavidKnott @wlffann After today's class and our conversation from last night around storing the full html in the localStorage I went back and re-did our localStorage setup. We are now storing the food items and their information in string formatted JSON. I've already updated the persisting items on page reloads. Also as I have a feeling we will have to continually plug the information into the table HTML I pulled that process out into its own function so all we will need to do in order to plug the information into the table is call
```formatFoodItemTable(name, calories);``` passing in the necessary information to the function. Let me know if you have any questions!

jdconrad89 committed:
Implement localStorage to allow us to have persisting items

jdconrad89 committed:
141617729 deletes unneeded functionality

As I was writing the PR comment I realized I didn’t need the clear text
function when I was reloading the page with the persisted Items

jdconrad89 committed:
141617729 create the loadTheFoodItem function

Pull
```
for (var i = 0; i < localStorage.length; i++){
    $("tbody").append(localStorage.getItem(localStorage.key(i)));
    add_delete_functionality();
  }
```
out into its own function

jdconrad89 committed:
141617729 Update the localStorage

After today’s class I went back and reset up the localStorage so that
it was purely holding on to a JSON string of the Name and the Calories.
I puled it out into multiple functions to limit the responsibilities of
each function.

jdconrad89 committed:
141617729 HTML Formatting Refactor

I extracted the html Formatting of the table out into it’s own function
since we are realistically going the using this quite a bit and it will
allow us to be more DRY in our coding

jdconrad89 committed:
14617729 Fix Merge Conflicts

I merged master down on my local branch and resolved the merge
conflicts. It was mostly spacing issues and me pulling pieces out into
their own function

wlffann commented on lib/foods.js:
```diff
@@ -15,66 +15,96 @@ $('document').ready(function() {
+
+function Food(name, calories) {
+  this.name = name;
+  this.calories = calories;
```
👍

wlffann commented on lib/foods.js:
```diff
@@ -15,66 +15,96 @@ $('document').ready(function() {
+  <td class='food-delete'><button>-</button></td>
+  </tr>`
+
+  $("tbody").prepend(foodRow);
```
Love this! I'm going to use it too.

wlffann reviewed (commented):
Looks great! Thanks for making these changes. This will be much easier to work with

DavidKnott commented on lib/foods.js:
```diff
@@ -14,13 +14,7 @@ $('document').ready(function() {
-    </tr>`
-
-    $("tbody").prepend(foodRow);
+    formatFoodItemTable(name, calories);
```
Great job extracting this into a method!

DavidKnott commented:
Looks great, after we merge in my PR would you want to squash some of your smaller commits?

jdconrad89 committed:
Fix merge conflicts so we can merge with master

The reason wasn’t working so I merged master onto my local branch and
fixed the conflicts and checked functionality. Again we need to double
check and see if contains works on Annie’s machine

DavidKnott commented:
Looks solid, I feel comfortable merging this

case-eee commented:
@jdconrad89 @DavidKnott @wlffann be very careful of your syntax! you're mixing snake case with camel case - i highly suggest fixing this before merging 👍

jdconrad89 committed:
set naming convention with camel case for consistency

jdconrad89 committed:
fix naming issue that was causing an error

jdconrad89 commented:
@case-eee the naming conventions have been fixed!
